compiler/transforms/accfg_config_overlap.py

```diff
@@ -171,6 +166,10 @@ def match_and_rewrite(self, op: accfg.SetupOp, rewriter: PatternRewriter, /):
             op.in_state.index - 1
         )  # -1 because the first block arg is the loop index
 
+        # also, if any operation between us and the loop start is not side effect free, abort
+        if any(not is_side_effect_free(prev_op) for prev_op in previous_ops_of(op)):
+            return
+
```

Comment:
So we can remove this now @AntonLydike

Reply:
Yes, it appears that we reached the conclusion that this was unneeded